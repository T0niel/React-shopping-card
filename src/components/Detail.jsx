import { ChevronDown, ChevronUp} from 'lucide-react';
import PropTypes from 'prop-types';
import { useState } from 'react';

export default function Detail({ text = '', children, maxHeightPixels = 200 }) {
  const [display, setDisplay] = useState(false);

  return (
    <div
      className=" bg-gray-200 cursor-pointer rounded"
      onClick={() => {
        setDisplay(!display);
      }}
    >
      <div className="flex items-center border-b-1 border-gray-300">
        <h1 className="p-2 mr-auto">{text}</h1>
        {display ? (
          <ChevronUp stroke="gray"></ChevronUp>
        ) : (
          <ChevronDown stroke="gray"></ChevronDown>
        )}
      </div>
      {display && (
        <div
          className="bg-gray-100 border-2 p-2 overflow-y-auto"
          style={{ maxHeight: `${maxHeightPixels}px` }}
        >
          {children}
        </div>
      )}
    </div>
  );
}

Detail.propTypes = {
  text: PropTypes.string,
  children: PropTypes.element,
  maxHeightPixels: PropTypes.number
};
