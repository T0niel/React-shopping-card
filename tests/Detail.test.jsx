import { describe, expect, it } from 'vitest';
import Detail from '../src/components/Detail';
import { render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Detail', () => {
  it('Match snapshot', () => {
    const { container } = render(<Detail text={''} ></Detail>);
    expect(container).toMatchSnapshot();
  });

  it('Displays the text', () => {
    render(<Detail text={'Hello'} ></Detail>);

    const text = screen.queryByText('hello');

    expect(text).not.toEqual(null);
  });

  it('Displays children onclick', () => {
    const user = userEvent.setup();

    const { container } = render(
      <Detail text={'test'} >
        <h1>some description</h1>
      </Detail>
    );

    user.click(container);
    expect(screen.queryByText('some description')).not.toEqual(null);
  });

  it('Hides children when clicked twice', () => {
    const user = userEvent.setup();

    const { container } = render(
      <Detail text={'test'}>
        <h1>some description</h1>
      </Detail>
    );

    user.click(container);
    expect(screen.queryByText('some description')).not.toEqual(null);
    user.click();
    expect(screen.queryByText('some description')).toEqual(null);
  });

  it('Does not display children when not clicked', () => {
    render(
      <Detail text={'test'}>
        <h1>some description</h1>
      </Detail>
    );

    expect(screen.queryByText('some description')).toEqual(null);
  });
});
