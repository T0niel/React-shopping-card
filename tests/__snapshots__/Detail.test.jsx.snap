// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Detail > Match snapshot 1`] = `
<div>
  <div
    class=" bg-gray-200 cursor-pointer rounded"
  >
    <div
      class="flex items-center border-b-1 border-gray-300"
    >
      <h1
        class="p-2 mr-auto"
      />
      <svg
        class="lucide lucide-chevron-down"
        fill="none"
        height="24"
        stroke="gray"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m6 9 6 6 6-6"
        />
      </svg>
    </div>
  </div>
</div>
`;
